### DISCUSSION TITLE
Next.JS: is it possible to use fetch to send a local audio .mp3 file (instead of a url) to DeepGram for a transcription?

### DISCUSSION THREAD

**Discussion by SteveK888** [deepgram/community#410] [General help] [answered]
Hi, I've been successful to get a transcription using Next.JS and fetch - but it only works by giving it the url to an audio file on the web.

Generally it's a Next.JS (version 13.5.6) await fetch version of the curl method at the Deepgram API Playground: https://playground.deepgram.com/?smart_format=true&language=en&model=nova-2-ea

But, wondering if these are possible:

1) Is it possible to send a local audio .mp3 file (instead of a url to a publicly accessible audio file), using a fetch?
What are the query parameters in the fetch? i.e. headers and body?

2) Is multipart/form-data allowed by Deepgram through a fetch?

3) experimenting with the CURL version of https://developers.deepgram.com/reference/pre-recorded
Where in the body I put the local .mp3 file in BODY PARAMS's RAW BODY, and the data changes in the REQUEST  EXAMPLES, however Try it! returns: 
{
  "err_code": "Bad Request",
  "err_msg": "Content-type was application/json, but we could not process the JSON payload.",
  "request_id": "50ac5b92-fb9c-44b2-b2ae-500ea52fbdae"
}

Your help is much appreciated.

**Comment by lukeocodes** [comment, marked as answer]
1. You can indeed! However, our API doesn't currently support custom CORS origins. We have a new JavaScript SDK being minted to beta tomorrow (so you can try the [alpha](https://www.npmjs.com/package/@deepgram/sdk/v/3.0.0-alpha.12) now) which will work in a client component on Next.js.

    In the [readme](https://github.com/deepgram/deepgram-node-sdk/tree/v3.0.0-alpha.12) for this version you'll find an explanation on the proxy support. You'll need a server component that can proxy the request to Deepgram. Any REST endpoints are subject to CORS restrictions. Effectively, you need to proxy the request to api.deepgram.com and return the results - you can also use this proxy to overwrite the Authorization header with the Deepgram API key, to prevent it from being exposed to the browser.  I hope to build a demo of this proxy soon, but it isn't on my immediate roadmap, sorry! 

    If you do manage to do it, I would love to showcase it on our communities.

    Alternatively, you could use this same SDK in a server component and make that request to Deepgram. Our old SDK can do this, and I have a demo of [doing it this way here](https://github.com/deepgram-devs/youtube-demo).


2. You send raw data, and you do not need to use a content-type.

    The problem you have is if you want to read a file from disk, you need a component on the server. If you deploy to Vercel or another serverless vendor, then they have no tmpfs (temporary filesystem) or any filesystem at all, so all `fs` functionality in Node is disabled. You cannot do `fs.createReadStream("./yourfile.mp3")`
    
    The alternative here is to use a URL. You send us the content-type of `application/json` and provide us the following request body `{ url: "public url to your file" }` and that will work just fine.
    
3. That example isn't brilliant and we hope to update it. I would really recommend using our Node (V2) SDK (in the youtube demo example) or experimenting with our new JavaScript (V3) SDK.

Hope this helps!

**Reply by SteveK888** [reply]
Thank you for the quick response and the answers, looking forward to putting them into action!
